missing receiver when method is passed as argument

### The following program `sample.go` triggers an unexpected result

```go
package main
 
type T struct {
    name string           
}                         
 
func (t *T) run(fn func(name string)) { fn(t.name) }
 
type T2 struct{}

func (t *T2) f(name string) { println("Hello", name) }                                          
                                                                                                
func main() {                                                                                   
    t := &T{"test"}                                                                             
    t2 := &T2{} 
    t.run(t2.f)                                                                                 
}
```


### Expected result

```console
$ go run ./sample.go
Hello test
```


### Got

```console
$ yaegi ./sample.go
./sample.go:7:44: panic
./sample.go:14:8: panic
run: reflect.Set: value of type string is not assignable to type *struct {}
goroutine 1 [running]:
runtime/debug.Stack()
	/Users/marc/sdk/go1.17.5/src/runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1(0x140001bbae0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/program.go:131 +0x8c
panic({0x1054e6d40, 0x14000114960})
	/Users/marc/sdk/go1.17.5/src/runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x140000d22c0, 0x14000133200, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/run.go:194 +0x164
panic({0x1054e6d40, 0x14000114960})
	/Users/marc/sdk/go1.17.5/src/runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x140000d24d0, 0x1400012ea20, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/run.go:194 +0x164
panic({0x1054e6d40, 0x14000114960})
	/Users/marc/sdk/go1.17.5/src/runtime/panic.go:1038 +0x21c
reflect.Value.assignTo({0x1054e6d40, 0x140001148e0, 0x198}, {0x1053029a1, 0xb}, 0x1400011a180, 0x0)
	/Users/marc/sdk/go1.17.5/src/reflect/value.go:2797 +0x2cc
reflect.Value.Set({0x1400011a180, 0x14000116358, 0x196}, {0x1054e6d40, 0x140001148e0, 0x198})
	/Users/marc/sdk/go1.17.5/src/reflect/value.go:1905 +0xa4
github.com/traefik/yaegi/interp.call.func9(0x140000d24d0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/run.go:1388 +0xb9c
github.com/traefik/yaegi/interp.runCfg(0x1400012ea20, 0x140000d24d0, 0x14000134480, 0x14000133c20)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/run.go:202 +0x268
github.com/traefik/yaegi/interp.call.func9(0x140000d22c0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/run.go:1399 +0x7c0
github.com/traefik/yaegi/interp.runCfg(0x14000133200, 0x140000d22c0, 0x140001327e0, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/run.go:202 +0x268
github.com/traefik/yaegi/interp.(*Interpreter).run(0x1400035c000, 0x140001327e0, 0x1400035e000)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/run.go:121 +0x3d4
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0x1400035c000, 0x14000111680)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/program.go:157 +0x1e0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x1400035c000, {0x140004b01e0, 0xe9}, {0x16b3937a0, 0x16}, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/interp.go:615 +0x98
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x1400035c000, {0x16b3937a0, 0x16})
	/Users/marc/go/src/github.com/traefik/yaegi-radix/interp/interp.go:508 +0xe8
main.runFile(0x1400035c000, {0x16b3937a0, 0x16}, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi-radix/cmd/yaegi/run.go:153 +0xb0
main.run({0x14000032050, 0x1, 0x1})
	/Users/marc/go/src/github.com/traefik/yaegi-radix/cmd/yaegi/run.go:116 +0xe14
main.main()
	/Users/marc/go/src/github.com/traefik/yaegi-radix/cmd/yaegi/yaegi.go:144 +0x384
```


### Yaegi Version

v0.11.1

### Additional Notes

Issue found when running `github.com/mediocregopher/radix/v4` in yaegi

--- a/_test/issue-1332.go
+++ b/_test/issue-1332.go
@@ -1,0 +1,17 @@
+package main
+
+func run(fn func(name string)) { fn("test") }
+
+type T2 struct {
+	name string
+}
+
+func (t *T2) f(s string) { println(s, t.name) }
+
+func main() {
+	t2 := &T2{"foo"}
+	run(t2.f)
+}
+
+// Output:
+// test foo

--- a/interp/run.go
+++ b/interp/run.go
@@ -1404,12 +1404,22 @@ func call(n *node) {
 					}
 				default:
 					val := v(f)
-					// The !val.IsZero is to work around a recursive struct zero interface
-					// issue. Once there is a better way to handle this case, the dest
-					// can just be set.
-					if !val.IsZero() || dest[i].Kind() == reflect.Interface {
-						dest[i].Set(val)
+					if val.IsZero() && dest[i].Kind() != reflect.Interface {
+						// Work around a recursive struct zero interface issue.
+						// Once there is a better way to handle this case, the dest can just be set.
+						continue
 					}
+					if nod, ok := val.Interface().(*node); ok && nod.recv != nil {
+						// An interpreted method is passed as value in a function call.
+						// It must be wrapped now, otherwise the receiver will be missing
+						// at the method call (#1332).
+						// TODO (marc): wrapping interpreted functions should be always done
+						// everywhere at runtime to simplify the whole code,
+						// but it requires deeper refactoring.
+						dest[i] = genFunctionWrapper(nod)(f)
+						continue
+					}
+					dest[i].Set(val)
 				}
 			}
 		}

